.github/workflows/_build-and-test-locally.yml

```diff
@@ -334,24 +344,127 @@ jobs:
+        env:
+          COMMIT_SHA: ${{ github.event.pull_request.head.sha || github.sha }}
+          BASE_SHA: ${{ github.event.pull_request.base.sha || github.sha }}
+        run: |
+          MAX_TESTS=3
+          CURRENT="${COMMIT_SHA}"
+          BASELINE="$(git merge-base $BASE_SHA $CURRENT)"
+          if [ -z $BASELINE ]; then
+            echo "Baseline commit for PR is not found, detection skipped."
+            exit 0
+          fi
+          true > /tmp/neon-pytest-new-tests.txt
+          true > /tmp/neon-pytest-updated-tests.txt
+          for t in $(git diff --name-only $BASELINE..$CURRENT test_runner/regress/); do
+            [ -f "$t" ] || continue
+            echo "Test file modified: $t"
+
+            # Get and compare two lists of items collected by pytest to detect new tests in the PR
+            { git show $BASELINE:"$t" || true; } > test_runner/regress/__temp__.py
+            { ./scripts/pytest --collect-only -q test_runner/regress/__temp__.py | sed '/^$/q; s/.*:://; s/\[.*//' | sort -u || true; } > /tmp/neon-pytest-tests-0.txt
+            cat /tmp/neon-pytest-tests-0.txt
+            { ./scripts/pytest --collect-only -q "$t" | sed '/^$/q; s/.*:://; s/\[.*//' | sort -u || true; } > /tmp/neon-pytest-tests-1.txt
+            cat /tmp/neon-pytest-tests-1.txt
+            comm -13 /tmp/neon-pytest-tests-0.txt /tmp/neon-pytest-tests-1.txt | sed "s|^|$t::|" >> /tmp/neon-pytest-new-tests.txt
+
+            # Detect pre-existing test functions updated in the PR
+            { git diff $BASELINE..$CURRENT "$t" | grep -o '^@@[0-9, +-]\+@@ def [^(]\+' | sed "s/.* def //" | sort -u | \
+              comm -12 /tmp/neon-pytest-tests-1.txt - || true; } > /tmp/neon-pytest-updated-functions.txt
+            cat /tmp/neon-pytest-updated-functions.txt
+            while read -r f; do
+              echo "Function modified: $f"
+              # Extract changes within the function
+              git log $BASELINE..$CURRENT -L ":$f:$t" >/tmp/neon-pytest-func-update.patch || continue
+
+              # Revert changes to get the file with only this function updated
+              # (applying the patch might fail if it contains a change for the next function declaraion)
+              cp "$t" test_runner/regress/__temp__.py
+              patch -R -i /tmp/neon-pytest-func-update.patch test_runner/regress/__temp__.py >/dev/null || continue
+
+              # Ignore whitespace-only changes
+              diff -w "$t" test_runner/regress/__temp__.py > /dev/null && continue
+
+              echo "$t::$f" >> /tmp/neon-pytest-updated-tests.txt
+            done < /tmp/neon-pytest-updated-functions.txt
+            rm test_runner/regress/__temp__.*
+          done
+          cat /tmp/neon-pytest-new-tests.txt /tmp/neon-pytest-updated-tests.txt | head -$MAX_TESTS > /tmp/neon-tests-to-test.txt
+          echo "tests=$(tr '\n' ' ' < /tmp/neon-tests-to-test.txt)" >> $GITHUB_OUTPUT
+
+      - name: Test new/updated pytest tests (v17-only)
```

Comment:
- New additions complicate the workflow pretty significantly; let's create a separate workflow for that and trigger it from the original one, maybe?

- I don't think that the failure of this should be blocking (at this point where we already have flaky tests).

Reply:
I added "continue-on-error: true" below, so PR should not be blocked when flaky tests are detected. Maybe I am wrong, but my experiments proved this.